lib/stdio/encoding.js

```diff
@@ -29,7 +29,7 @@ const encodingEndGenerator = async function * (encoding, chunks) {
 	const stringDecoder = new StringDecoder(encoding);
 
 	for await (const chunk of chunks) {
-		yield stringDecoder.write(chunk);
+		yield isUint8Array(chunk) ? stringDecoder.write(chunk) : chunk;
```

Comment:
This is the transform used for the `encoding: 'base64'` option.
Since previous transforms might use `objectMode: true` and return any type, chunks of this transform might be of any type. Encoding non-strings/`Uint8Array` to base64 becomes a noop then.